### ISSUE TITLE
'<<<>>>' appearing in range with zh_TW locale

### ISSUE THREAD

**Issue by campbell-m** [flack/ranger#15]
I've modified my test program slightly:

~~~php
<?php
use OpenPsa\Ranger\Ranger;

require_once 'lib/autoload.inc';

error_reporting(-1);
ini_set('display_errors', '1');

foreach (['en_GB', 'zh_TW'] as $locale) {
    $ranger = new Ranger($locale);
    $ranger
        ->setDateType(IntlDateFormatter::MEDIUM)
        ->setTimeType(IntlDateFormatter::SHORT);
    echo $ranger->format(1711738800, 1711828800);
    echo "<br>\n";
}
~~~

and now get the following output:

~~~text
29 Mar 2024, 19:00 – 30 Mar 2024, 20:00
2024年3月29日, 晚上<<<>>>7:00 – 2024年3月30日, 晚上<<<>>>8:00
~~~

The problem only seems to happen if the start and end dates are on different days.

**Comment by campbell-m** [comment]
PS Running on 0.5.8.

**Comment by campbell-m** [comment]
And running on a Linux system with PHP 8.3.2 I get

~~~text
29 Mar 2024, 19:00 – 30 Mar 2024, 20:00

Fatal error: Uncaught RuntimeException: missing separator between date parts in /home/bfdaeeca/public_html/book/lib/OpenPsa/Ranger/Ranger.php:419 Stack trace: #0 /home/bfdaeeca/public_html/book/lib/OpenPsa/Ranger/Ranger.php(182): OpenPsa\Ranger\Ranger->parse_pattern() #1 /home/bfdaeeca/public_html/book/test.php(14): OpenPsa\Ranger\Ranger->format() #2 {main} thrown in /home/bfdaeeca/public_html/book/lib/OpenPsa/Ranger/Ranger.php on line 419
~~~

I can't work out what the difference is between the two systems that's causing one to throw an exception.

**Comment by flack** [comment]
@campbell-m for the first issue I commited a fix (6896106f1a5e74a672beb0fadc41615c8ff6d1bb), will take a look at the second one later

**Comment by flack** [comment]
for your second issue: Are you sure you're on the current version? Because that runtime exception was removed with the fix for #14. Line 419 looks like this now: https://github.com/flack/ranger/blob/master/src/Ranger.php#L419

**Comment by campbell-m** [comment]
Many thanks for the fix, which solves the problem.  And yes, apologies, I was using an old version of Ranger for the second issue.
